client/src/controllers/LoaderController.ts

```diff
@@ -0,0 +1,130 @@
+import { Application, Controller } from '@hotwired/stimulus';
+import { gettext } from '../utils/gettext';
+
+/**
+ * <button
+ *   type="submit"
+ *   class="button button-longrunning"
+ *   data-controller="w-loader"
+ *   data-w-loader-active-class="button-longrunning-active"
+ *   data-w-loader-active-value="{% trans 'Signing in…' %}"
+ *   data-w-loader-duration-seconds-value="40"
+ *   data-action="w-loader#activate"
+ * >
+ *  {% icon name="spinner" %}
+ *  <em data-w-loader-target="label">{% trans 'Sign in' %}</em>
+ * </button>
+ */
+export class LoaderController extends Controller {
+  static classes = ['active'];
+  static targets = ['label'];
+  static values = {
+    loading: { type: Boolean, default: false },
+    active: { type: String, default: gettext('Loading') },
+    durationSeconds: { type: Number, default: 30 },
+    label: { type: String, default: '' },
+  };
+
+  declare activeClass: string;
+  declare activeValue: string;
+  declare loadingValue: boolean;
+  declare durationSecondsValue: number;
+  declare labelTarget: HTMLElement;
+  declare hasActiveClass: boolean;
+  declare hasLabelTarget: boolean;
+  declare labelValue: string;
+  timer?: number;
+
+  static afterLoad(identifier: string, application: Application) {
+    const { controllerAttribute } = application.schema;
+    const { actionAttribute } = application.schema;
+
+    const updateLoader = () => {
+      const em = document.querySelector('em') as HTMLElement;
+      document
+        .querySelectorAll(
+          `.button-longrunning:not([data-controller~='${identifier}'])`,
+        )
+        .forEach((button) => {
+          button.setAttribute(controllerAttribute, identifier);
+          button.setAttribute(actionAttribute, 'w-loader#activate');
+
+          if (button.getAttribute('data-clicked-text')) {
+            button.setAttribute(
+              'data-w-loader-active-value',
+              `${button.getAttribute('data-clicked-text')}`,
+            );
+            button.removeAttribute('data-clicked-text');
+          }
+
+          if (em.matches('button em')) {
+            em.setAttribute('data-w-loader-target', 'label');
+          }
+
+          button.setAttribute('data-w-loader-duration-seconds-value', '10');
+        });
+    };
+
+    document.addEventListener('DOMContentLoaded', updateLoader);
+  }
```

Comment:
```suggestion
  static afterLoad(identifier: string, application: Application) {
    const { controllerAttribute } = application.schema;
    const { actionAttribute } = application.schema;

    document.addEventListener('DOMContentLoaded', () => {
      document
        .querySelectorAll(
          `.button-longrunning:not([${controllerAttribute}~='${identifier}'])`,
        )
        .forEach((button) => {
          button.setAttribute(controllerAttribute, identifier);
          button.setAttribute(actionAttribute, `${identifier}#activate`);

          const activeText = button.getAttribute('data-clicked-text');
          if (activeText) {
            button.setAttribute(`data-${identifier}-active-value`, activeText);
            button.removeAttribute('data-clicked-text');
          }

          const labelElement = button.querySelector('em');
          if (labelElement) {
            labelElement.setAttribute(`data-${identifier}-target`, 'label');
          }

          button.setAttribute(
            `data-${identifier}-duration-seconds-value`,
            '30',
          );
        });
    });
  }
```

A few refinements
* ensure we use the existing default 30 second count
* be careful with the `em` selector - this was grabbing all `em` elements on the entire page, not the one within the button
* Update all usage of `w-loader` string with the actual variable `identifier` - this way when you update the controller identifier in one place (index.ts) you do not need to update this place either. e.g. changing to `w-progress`